ArchiSteamFarm/Bot.cs

```diff
@@ -1185,15 +1185,14 @@ public sealed class Bot : IDisposable {
 					partLength = message.Length - i;
 				}
 
-				string messagePart = message.Substring(i, partLength);
-
 				// If our message is of max length and ends with a single '\' then we can't split it here, it escapes the next character
-				if ((messagePart.Length >= maxMessageLength) && (messagePart[messagePart.Length - 1] == '\\') && (messagePart[messagePart.Length - 2] != '\\')) {
+				if ((partLength == maxMessageLength) && (message[i + partLength - 1] == '\\') && (message[i + partLength - 2] != '\\')) {
```

Comment:
Why did you change the operator? `>=` was there for a reason.

Reply:
Because partLength can't be more than maxMessageLength.

Reply:
So? `maxMessageLength` isn't the max value for the variable type, same goes for comparing stuff to `0`, if it's `int` then you should compare with `<= 0`, even if it's never gonna be into negative, unless you have `uint`, in which case you should compare to `0`. Avoid exact matches unless you mean exact match and value above/below exact match is inappropriate, cover more cases than less.

Reply:
Well, partLength>maxMessageLength is inappropriate, because it would mean that whole thing is fucked up, as steam won't be able to send the message anyway. But I may add another check that would throw critical exception in this case, that would be a correct approach, because it would mean that RAM is corrupted.

Reply:
No need to check against things that can't happen, this is a safeguard for the future that doesn't affect runtime, performance or anything. Current code is OK.